src/easylink/graph_components.py

```diff
@@ -87,27 +170,58 @@ def add_edge_from_params(self, edge_params: EdgeParams) -> None:
 
 
 class ImplementationGraph(nx.MultiDiGraph):
-    """A graph of Implementations, with edges representing file dependencies between them.
+    """A graph of :class:`Implementations<easylink.implementation.Implementation>`.
+
+    ImplementationGraphs are directed graphs with :class:`Implementations<easylink.implementation.Implementation>`
+    for nodes and the file dependencies between them for edges. Self-edges as well
+    as multiple edges between nodes are permitted.
+
+    Notes
+    -----
+    Unlike a :class:`StepGraph`, an ImplementationGraph is a low-level abstraction;
```

Comment:
I'm fine with "low-level" and "high-level" but I think a bit more detail could be helpful.

For example: Unlike a StepGraph, which can be a complex nested structure due to the various complex and self-similar Step classes (which represent abstract operations such as "loop this step N times"), an ImplementationGraph is a flat, concrete graph of Implementations to run, with abstract operations unrolled.